extensions/amp-base-carousel/0.1/test/test-carousel.js

```diff
@@ -236,4 +236,30 @@ describes.realWin('carousel implementation', {}, (env) => {
       expect(carousel.isAtStart()).to.be.false;
     });
   });
+
+  describe('resetScrollReferencePoint_', () => {
+    it('currentElementOffset_ & currentIndex_ should be set when it is a' + 
+    ' programmatic scroll', async () => {
+      setStyle(element, 'width', '299.2px');
+
+      const carousel = await createCarousel({
+        slideCount: 12,
+        loop: false,
+      });
+
+      // Fake the scroll that ends short of the correct index.
+      // This is handled by scroll event listener.
+      carousel.touching_ = false;
```

Comment:
Thank you for adding a comment here describing what the intent of these values are, nice touch!